boot.js

```diff
@@ -350,7 +350,10 @@ function closeWithCbOrNextTick (func, cb, context) {
 function encapsulateTwoParam (func, that) {
   return _encapsulateTwoParam.bind(that)
   function _encapsulateTwoParam (context, cb) {
-    if (func.length === 0 || func.length === 1) {
+    if (func.length === 0) {
+      func()
+      process.nextTick(cb)
+    } else if (func.length === 1) {
       func(this)
       process.nextTick(cb)
```

Comment:
You were right. This needs to change to:

```js
func(cb)
```

Remove the ` process.nextTick` thing.

Reply:
Tests are failing with that change `not ok child test left in queue:`
```js
function encapsulateTwoParam (func, that) {
  return _encapsulateTwoParam.bind(that)
  function _encapsulateTwoParam (context, cb) {
    if (func.length === 0) {
      func()
      process.nextTick(cb)
    } else if (func.length === 1) {
      func(cb)
    } else {
      func(this, cb)
    }
  }
}
```

Reply:
Maybe tests need to be updated to call `done()` ?